src/services/codefixes/convertTypedefToType.ts

```diff
@@ -0,0 +1,249 @@
+    const { typeExpression } = tag;
+    if (!typeExpression) return;
+    const typeName = tag.name?.getFullText().trim();
+    if (!typeName) return;
+
+    // For use case @typedef {object}Foo @property{bar}number
+    // But object type can be nested, meaning the value in the k/v pair can be object itself
+    if (typeExpression.kind === SyntaxKind.JSDocTypeLiteral) {
+        return createInterfaceForTypeLiteral(typeName, typeExpression);
+    }
+    // for use case @typedef {(number|string|undefined)} Foo or @typedef {number|string|undefined} Foo
+    // In this case, we reach the leaf node of AST.
+    // Here typeExpression.type is a TypeNode, e.g. a UnionType or a primitive type, such as "number".
+    if (typeExpression.kind === SyntaxKind.JSDocTypeExpression) {
+        return createTypeAliasForTypeExpression(typeName, typeExpression);
+    }
+}
+
+function createInterfaceForTypeLiteral(
+    typeName: string,
+    typeLiteral: JSDocTypeLiteral
+): InterfaceDeclaration | undefined {
+    const propertySignatures = createSignatureFromTypeLiteral(typeLiteral);
+    if (!propertySignatures || propertySignatures.length === 0) return;
+    const interfaceDeclaration = factory.createInterfaceDeclaration(
+        [],
+        typeName,
+        [],
+        [],
+        propertySignatures,
+    );
+    return interfaceDeclaration;
+}
+
+function createTypeAliasForTypeExpression(
+    typeName: string,
+    typeExpression: JSDocTypeExpression
+): TypeAliasDeclaration | undefined {
+    const typeReference = createTypeReference(typeExpression.type);
+    if (!typeReference) return;
+    const declaration = factory.createTypeAliasDeclaration(
+        [],
+        factory.createIdentifier(typeName),
+        [],
+        typeReference
+    );
+    return declaration;
+}
+
+function createSignatureFromTypeLiteral(typeLiteral: JSDocTypeLiteral): PropertySignature[] | undefined {
+    const propertyTags = typeLiteral.jsDocPropertyTags;
+    if (!propertyTags || propertyTags.length === 0) return;
+
+    const getSignatures = (signatures: PropertySignature[], tag: JSDocPropertyTag) => {
+        const name = getPropertyName(tag);
+        const type = tag.typeExpression?.type;
+        let typeReference;
+
+        // Recursively handle nested object type
+        if (type && type.kind === SyntaxKind.JSDocTypeLiteral) {
+            const signatures = createSignatureFromTypeLiteral(type as JSDocTypeLiteral);
+            typeReference = factory.createTypeLiteralNode(signatures);
+        }
+        // Leaf node, where type.kind === SyntaxKind.JSDocTypeExpression
+        else if (type) {
+            typeReference = createTypeReference(type);
+        }
+        if (typeReference && name) {
+            const prop = factory.createPropertySignature(
+                [],
+                name,
+                // eslint-disable-next-line local/boolean-trivia

```

Comment:
You should fix this lint rather than hiding it; e.g. put `/*nameOfParameter*/` before `undefined` to indicate what parameter it's intending to be.

Reply:
fixed